File: Simple-NN-library/C++/src/networks/node.cpp
#include <networks/node.h>

using namespace the_network;
using namespace networks;
using namespace functions;

Node::Node(float weights[], float biasses[], float* activation_function){
  this.set_weights(weights);
  this.set_biasses(biasses);
  this.set_activation(activation_function);
}
Node::~Node(){

}

float Node::formula(float calculation_array[][3]){
  float total=0;
  for(int i=0; i<sizeof(calculation_array); i++){
    total+=(calculation_array[i][0]*calculation_array[i][1])+calculation_array[i][2];
  }
  return total;
}

float Node::calculate(float inputs[]){
  if(sizeof(inputs)!=sizeof(this.weights)||sizeof(inputs)!=sizeof(this.biasses)){
    return (float)NULL;
  }
  float calculation_array[sizeof(inputs)][3];
  for(int i=0; i<sizeof(inputs); i++){
    calculation_array[i]={inputs[i], this.weights[i], this.biasses[i]};
  }

  float calculation_result=sum(this.formula(calculation_array));
  return this.activation_function(calculation_result);
}

void Node::set_weights(float weights[]){
  this.weights=weights;
}
void Node::set_biasses(float biasses[]){
  this.biasses=biasses;
}
bool Node::set_activation(float* activation_function){
  for(int i=0; i<sizeof(Activation_functions.list_of_functions); i++){
    if(Activation_functions.list_of_functions[i]==activation_function){
      this.activation_function=activation_function;
      return true;
    }
  }
  return false;
}

File: Simple-NN-library/python/networks/node.py
from .further_maths import*
from functions.activation_functions import*

class Node:
    def __init__(self, weights, biasses, activation_function):
        self.set_weights(weights)
        self.set_biasses(biasses)
        self.set_activation(activation_function)

    def formula(self, calculation_array):
        total=0
        for i in calculation_array:
            total+=(i[0]*i[1])+i[2]
        return total

    def calculate(self, inputs):
        if(len(inputs)!=len(self.weights) or len(inputs)!=len(biasses)):
            return None

        calculation_array=[]
        for i in range(len(inputs)):
            calculation_array[i]=[inputs[i], self.weights[i], self.biasses[i]]

        calculation_result=sum(self.formula(calculation_array))
        return self.activation_function(calculation_result)

    def set_weights(self, weights):
        self.weights=weights

    def set_biasses(self, biasses):
        self.biasses=biasses

    def set_activation_function(self, activation_function):
        for i in range(len(Activation_functions.list_of_functions)):
            if(Activation_functions.list_of_functions[i]==activation_function):
                self.activation_function=activation_function
                return True
        return False
